crates/router/src/core/files.rs

```diff
@@ -0,0 +1,104 @@
+#[cfg(feature = "s3")]
+pub mod s3_utils;
+pub mod transformers;
+
+#[cfg(not(feature = "s3"))]
+pub mod fs_utils;
+
+use api_models::files;
+use error_stack::{IntoReport, ResultExt};
+
+use super::errors::{self, RouterResponse};
+use crate::{
+    consts,
+    routes::AppState,
+    services::{self, ApplicationResponse},
+    types::{api, storage},
+};
+
+pub async fn files_create_core(
+    state: &AppState,
+    merchant_account: storage::merchant_account::MerchantAccount,
+    create_file_request: api::CreateFileRequest,
+) -> RouterResponse<files::CreateFileResponse> {
+    transformers::validate_file_upload(
+        state,
+        merchant_account.clone(),
+        create_file_request.clone(),
+    )
+    .await?;
+    let file_id = common_utils::generate_id(consts::ID_LENGTH, "file");
+    let file_key = format!("{}_{}", merchant_account.merchant_id, file_id);
```

Comment:
Prefer using `/` as delimiters (as opposed to `_`) when it comes to S3 keys to take advantage of S3's folder-like namespacing which makes it easier to view in the UI and makes things like `Resource` specifications in presigned URL policies more intuitive.

Reply:
Also just having merchant ID and file ID in the key kinda makes it too shallow. Probably better to have something like `{merchant_id}/{dispute_id}/{file_id}`